[WIP] Dev/explorer add formatting types

linus-sherrill opened the pull request:
Expand explorer plugin architecture. Explorer plugins provide a way to generate custom formatting
for specific plugin categories. Currently only `process` and `algorithm` are supported. Plugin categories are stored in an attribute of the plugin factory object.

A new `--fmt xxx` command line option is added to specify an  alternate output format. If the `--fmt`
option is not specified, the factory category name is used to select the formatting plugin. For example, the explorer plugin named "process" is used to format factories that have the "process" category. If the `--fmt xxx` command line option is present, plugin explorer will use the "process-xxx" plugin for format processes. This provides a flexible way to extend the plugin formatting operation.

Added support for generating process documentation in rst and pipe file format.
Added support for generating algorithm documentation in pipeline format.

aaron-bray commented:
I am still wrapping my head around the code.
Generally can you take out the title block?
for example:
```
frame_list_input
================
```
I am going to still have a rst file for the frame_list_input that will have the title that we can use to add context to this process, then directly inline include the file(s) that the process explorer will make.

This looks so good, I don't think we need 2 files anymore, we just need to decide order between Configuration,Pipefile Usage, and Class Description

I had to add a return right after the csv table and the next 'Input Port' section
There has to be an empty line between sections

add the :members: do the doxygen so we get public methods
(unless you don't want to show those)
```
.. doxygenclass:: kwiver::frame_list_process
   :project: kwiver
   :members:
```
 In the pipe sections can you name the process the same either <this-name> or <this-proc>, i kind of like <this-proc> better.

And there is a '<br>' getting into the rst file
`
Reads a list of image file names and generates stream of images and associated<br>time stamps
`

linus-sherrill commented:
I removed the process name header. Also combined output into one file. I like the current order of sections. config & ports tables, usage information. I added the :members: item to the doxygen section, but not sure it helps any for processes, since they have so few members.

aaron-bray commented:
Cool, do you want to merge this into the doc-updates branch before the master?

linus-sherrill commented:
I merged my branch into doc-updates and will cancel the other PR

On Fri, Oct 27, 2017 at 11:44 AM, Aaron Bray <notifications@github.com>
wrote:

> Cool, do you want to merge this into the doc-updates branch before the
> master?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/kwiver/pull/344#issuecomment-340008413>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGl1PEZSM7jgbrEis8PgFMzTqO_lkVwwks5swfpNgaJpZM4QISWe>
> .
>



-- 
*Linus Sherrill - *Staff R&D Engineer
Kitware, Inc.
28 Corporate Drive
Clifton Park, NY 12065-8662
E: linus.sherrill@kitware.com
P: 518.881.4400

linus-sherrill commented:
This branch has been merged to another branch then to master.
